src/components/PeriodDimension/FixedPeriodSelect.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react'
+import propTypes from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { SingleSelectField, SingleSelectOption } from '@dhis2/ui'
+import FixedPeriodFilter from './FixedPeriodFilter.js'
+import {
+    parsePeriodCode,
+    getFixedPeriodsOptionsById as getPeriodById,
+    getYearOffsetFromNow,
+} from './utils/fixedPeriods.js'
+import styles from './styles/FixedPeriodSelect.style.js'
+
+const FixedPeriodSelect = ({
+    onChange,
+    allowedPeriodTypes,
+    className,
+    dataTest,
+    value,
+}) => {
+    const [periodType, setPeriodType] = useState('')
+    const [year, setYear] = useState('')
+    const [options, setOptions] = useState()
+
+    useEffect(() => {
+        const period = parsePeriodCode(value, allowedPeriodTypes)
+        if (period) {
+            setPeriodType(period.id)
+            setYear(period.year)
+            setOptions(period.options)
+        } else if (periodType && year) {
```

Comment:
I actually think this case will NEVER run if `value` and `allowedPeriodTypes` are passed, since `periodType` and `year` are derived from those?